nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/iot/util/IoTMessage.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.aws.iot.util;
+
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+
+public class IoTMessage {
+    private final String topic;
+    private final byte[] payload;
+    private final Integer qos;
+
+    public IoTMessage(MqttMessage message, String topic) {
```

Comment:
**JPercivall**
MqttMessage also offers the "isDuplicate" and "isRetained" methods. Is there a reason for not including these as part of IoTMessage and passing onto the FlowFile as attributes?